### ISSUE TITLE
json path does not support `last` and `range`

### ISSUE THREAD

**Issue by xiongjiwei** [pingcap/tidb#37289]
## Enhancement
```sql
SELECT JSON_ARRAY_INSERT(@json, '$.A[last-1][last-1]', 5);
SELECT JSON_ARRAY_INSERT(@json, '$.A[2 to 5]', 5);
```

tidb does not support it

**Comment by YangKeao** [comment]
It seems that this grammar is also not supported in MySQL 5.7:

```
mysql> SET @json = '{"A":[1,2,3,4,5,6]}';
Query OK, 0 rows affected (0.00 sec)

mysql> SELECT JSON_ARRAY_INSERT(@json, '$.A[2 to 5]', 5);
ERROR 3143 (42000): Invalid JSON path expression. The error is around character position 7.
mysql> SELECT JSON_ARRAY_INSERT(@json, '$.A[last-1][last-1]', 5);
ERROR 3143 (42000): Invalid JSON path expression. The error is around character position 4.
```

**Comment by xiongjiwei** [comment]
it's new for mysql8, we can support it in the future

**Comment by dveeden** [comment]
@YangKeao isn't this fixed in #38584 ?

This is working in TiDB now:
```
mysql> SELECT JSON_EXTRACT('["a","b","c","d"]','$[2 to 5]');
+-----------------------------------------------+
| JSON_EXTRACT('["a","b","c","d"]','$[2 to 5]') |
+-----------------------------------------------+
| ["c", "d"]                                    |
+-----------------------------------------------+
1 row in set (0.00 sec)
```

**Comment by YangKeao** [comment]
> @YangKeao isn't this fixed in #38584 ?
> 
> This is working in TiDB now:
> 
> ```
> mysql> SELECT JSON_EXTRACT('["a","b","c","d"]','$[2 to 5]');
> +-----------------------------------------------+
> | JSON_EXTRACT('["a","b","c","d"]','$[2 to 5]') |
> +-----------------------------------------------+
> | ["c", "d"]                                    |
> +-----------------------------------------------+
> 1 row in set (0.00 sec)
> ```

Yes! This issue can be closed now :+1: . Thanks.
